cEP-0005.md

```diff
@@ -102,18 +102,25 @@ class RedundancyBear(LocalBear):
     # This bear is supposed to be able to fix unused imports...
     FIX_ASPECTS = {Root.Redundancy.UnusedImport}
     # ... and detect code clones.
-    DETECT_ASPECTS = {Root.Redundancy.CloneRedundancy}
+    DETECT_ASPECTS = {Root.Redundancy.Clone}
 
     # Aspects are passed as parameter
     def run(self, filename, file, aspects):
-        # No documentation required anymore for the bears and their parameters.
+        # No documentation required anymore for the bears.
 
-        # Check the code ...
+        if unused_imports(file):
+            # A bear may yield results for any aspect even if it's not selected.
+            # coala will filter out only selected aspects.
+            yield Result.from_values(aspect=Root.Redundancy.UnusedImprt, ...)
 
-        # A bear may yield results for any aspect even if it's not selected but
-        # may use the selected aspects to save performance. coala will filter
-        # out only selected aspects.
-        yield Result.from_values(aspect=Root.Redundancy.Clone, ...)
+        # Bears can save performance by only performing needed checks
+        if aspects[Root.Redundancy.Clone].active:
+            # Bears can access settings right from the aspect instance
+            min_clone_tokens = aspects[Root.Redundancy.Clone].min_clone_tokens
+
+            # The aspect is tied to the result. coala now knows everything from
+            # the aspect documentation!
+            yield Result.from_values(aspect=Root.Redundancy.Clone, ...)
```

Comment:
I think in the usual case this won't be possible easily because of shared logic, we might however include that possibility in the bear base class at a later point. This is however nothing that changes the main design so I wouldn't consider it yet especially as it probably comes with a performance penalty at least in most possible usecases (due to shared code)